pkg/controller/nodeipam/legacyprovider.go

```diff
@@ -62,7 +63,8 @@ func createLegacyIPAM(
 	}
 	ipamc, err := ipam.NewController(cfg, kubeClient, cloud, cidr, serviceCIDR, nodeCIDRMaskSizes[0])
 	if err != nil {
-		klog.Fatalf("Error creating ipam controller: %v", err)
+		klog.Warningf("Error creating ipam controller: %v", err)
+		panic(fmt.Sprintf("Error creating ipam controller: %v", err))
```

Comment:
@bowei @sarveshr7 what do you think to replace klog.Fatal from a. panic, the difference AFAIK is that the former does an `os.Exit`

Reply:
I think it would be better to return an error from this function and make all callers handle the error... test callers can fail the test or suppress the error, non-test callers should fail init.

A panic risks a non-test caller recovering and continuing on in a broken state

Reply:
agree, it seems the error is handling correctlly by the callers and fail to start the corresponding controller-manager

Reply:
Let's fix this to bubble up the error.